libs/base/Data/String.idr

```diff
@@ -49,24 +50,29 @@ fastUnlines = fastConcat . unlines'
 ||| Splits a character list into a list of whitespace separated character lists.
 |||
 ||| ```idris example
-||| words' (unpack " A B C  D E   ")
+||| words' (unpack " A B C  D E   ") [<] [<]
 ||| ```
-covering
-words' : List Char -> List (List Char)
-words' s = case dropWhile isSpace s of
-            [] => []
-            s' => let (w, s'') = break isSpace s'
-                  in w :: words' s''
+words' :  List Char
+       -> SnocList Char
+       -> SnocList (List Char)
+       -> List (List Char)
+words' (c :: cs) sc css =
+  if isSpace c
+     then case sc of
+       [<] => words' cs [<] css
+       _   => words' cs [<] (css :< (sc <>> Nil))
+     else words' cs (sc :< c) css
+words' [] [<] css = css <>> Nil
+words' [] sc  css = (css :< (sc <>> Nil)) <>> Nil
```

Comment:
Should this also case on whether `sc` is empty?

Reply:
I extracted a `wordsHelper` function to do what you described above.